### PR TITLE
Fix: Authentication Issue: azure.azcollection.azure_rm_storageblob Fails with Disabled Shared Keys

### PR DESCRIPTION
**Pull request by paronbol**
##### SUMMARY
PR in order to fix the feature of accessing storage blobs without SAS/key.
It is related with: #1255 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
- azure.azcollection.azure_rm_storageblob
- plugins/module_utils/azure_rm_common.py

##### ADDITIONAL INFORMATION
Now the access without SAS is supported by environment variables and cli

### CODE REVIEW

File: plugins/module_utils/azure_rm_common.py [hunk 1]
```diff
@@ -702,6 +702,12 @@ def get_blob_service_client(self, resource_group_name, storage_account_name, aut
             account = self.storage_client.storage_accounts.get_properties(resource_group_name=resource_group_name, account_name=storage_account_name)
             if auth_mode == 'login' and self.azure_auth.credentials.get('credential'):
                 credential = self.azure_auth.credentials['credential']
+            elif (auth_mode == 'login' and self.azure_auth.credentials.get('tenant')
+                  and self.azure_auth.credentials.get('client_id')
```
Comment (Fred-sun) [inline plugins/module_utils/azure_rm_common.py]: remove whitespace
```suggestion
                  and self.azure_auth.credentials.get('client_id')
```

File: plugins/module_utils/azure_rm_common.py [hunk 2]
```diff
@@ -702,6 +702,12 @@ def get_blob_service_client(self, resource_group_name, storage_account_name, aut
             account = self.storage_client.storage_accounts.get_properties(resource_group_name=resource_group_name, account_name=storage_account_name)
             if auth_mode == 'login' and self.azure_auth.credentials.get('credential'):
                 credential = self.azure_auth.credentials['credential']
+            elif (auth_mode == 'login' and self.azure_auth.credentials.get('tenant')
+                  and self.azure_auth.credentials.get('client_id') 
+                  and self.azure_auth.credentials.get('secret')):
+                credential = client_secret.ClientSecretCredential(tenant_id=self.azure_auth.credentials.get('tenant'),
```
Comment (Fred-sun) [inline plugins/module_utils/azure_rm_common.py:708]: continuation line under-indented for visual indent
```e.g.
                credential = client_secret.ClientSecretCredential(tenant_id=self.azure_auth.credentials.get('tenant'),
                                                                  client_id=self.azure_auth.credentials.get('client_id'),
                                                                  client_secret=self.azure_auth.credentials.get('secret'))

**Review by Fred-sun — COMMENTED** [review]
@paronbol This certainly is not reasonable, you through the client_secret. ClientSecretCredential is not authorized to obtain credentials, cannot manage a blob of resources. Thanks!

**Comment by paronbol** [comment]
> @paronbol This certainly is not reasonable, you through the client_secret. ClientSecretCredential is not authorized to obtain credentials, cannot manage a blob of resources. Thanks!

Hi @Fred-sun, thank you for your support.



I propose to use client_secret.ClientSecretCredential() because is exactly the same method used in the working AzureCLI credentials method for the api rest in get_mgmt_svc_client(). 
Even is not required to import it.
It is used in:
https://github.com/ansible-collections/azure/blob/5191b948716f92a01ea7f7ff3c58fd5274e1e507/plugins/module_utils/azure_rm_common.py#L1505-L1510

[MS Documentation of the ClientSecretCredential Class](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.clientsecretcredential?view=azure-python)

[Source Code of the class](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/client_secret.py)


I tried on a real blob storage and worked. The class is able to obtain credentials to manage blob resources.
But maybe you need the properly configure RBAC right to access through Microsft Entra ID, in my case is a Service Principal.
[More info in MS documentation](https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory#assign-azure-roles-for-access-rights)

MS for security recommends to disable SAS and use login or shared key when it is not possible. This is the importance of supporting this feature.

**Comment by Fred-sun** [comment]
@paronbol What you refer to configuring RBAC access, can you be more specific? I don't quite understand the authorization you said, thank you!

**Comment by paronbol** [comment]
In order to provide access to Service Principal into a Blob Storage you need to provide data access using a role, e.g. Storage Blob Data Contributor role
[More info in MS documentation](https://learn.microsoft.com/en-us/azure/storage/blobs/assign-azure-role-data-access?tabs=portal)

**Comment by Fred-sun** [comment]
@paronbol I assigned the role, but the mistake is still, I am investigating the reason, thank you!

**Review by xuzhang3 — APPROVED** [review]

**Comment by Ratanavery** [comment]
Hi, I am facing an issue when trying to use "auth_source: msi" for downloading the azure storage blob.

Looks like it always defaults to storage account key based access and when i use "auth_mode: login", it asks for msgraph module (where I think it will try to login) but msi doesn`t need to do any login.

PS:
I am able to use the azure key vault without any issues using the MSI.

**Comment by Fred-sun** [comment]
@Ratanavery Thank for your feedback! Welcome to submit your problems!

**Comment by xuzhang3** [comment]
@Ratanavery  Which value did you set for `auth_mode`? The default is `key`

**Pull request merged**